src/document-submit.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {user} from '../src/log';
+import {assertHttpsUrl} from '../src/url';
+
+
+/**
+ * @param {!Window} window
+ */
+export function installGlobalSubmitListener(window) {
+  new SubmitHandler(window)
+}
+
+
+/**
+ * Intercept any submit on the current document and polyfill validation
+ * check for the requested form.
+ */
+export class SubmitHandler {
+  /**
+   * @param {!Window} window
+   */
+  constructor(window) {
+    /** @private @const {!Window} */
+    this.win = window;
+
+    /** @private @const {!function(!Event)|undefined} */
+    this.boundHandle_ = this.handle_.bind(this);
+    this.win.document.documentElement.addEventListener(
+        'submit', this.boundHandle_, true);
+  }
+
+  /**
+   * Removes all event listeners.
+   */
+  cleanup() {
```

Comment:
There is no caller of this. Can this entire class just be inlined into the function `installGlobalSubmitListener`